### PR TITLE
fix: without email speaker going to my-session

### LINKED ISSUE #5342: Organizer: Adding speaker without email results in sessions added to "my sessions" from organizer him/herself
If an organizer adds a speaker "without email" the overview page shows the email of the organizer and sessions show up under "my sessions".

Screenshot from 2020-10-21 14-26-34

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #5342 
email is also required relation like track, so both needs to be make optional on server. That's why setting newSpeaker 'email' to " " .
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/events/view/sessions/create.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default class extends Controller {
     if (this.addNewSpeaker) {
       const newSpeaker = this.model.speaker;
       if (newSpeaker.isEmailOverridden) {
-        newSpeaker.set('email', this.authManager.currentUser.email);
+        newSpeaker.set('email', " ");
```
Comment (iamareebjamal) [inline app/controllers/events/view/sessions/create.js:11]: null

**Comment by iamareebjamal** [comment]
Is it working correctly?

**Review by NaviRocker — CHANGES_REQUESTED** [review]
Squash your Commits

**Comment by mariobehling** [comment]
To ensure we understand the feature in the same way:
1. Email is always required for speakers filling in the call for speakers.
2. This can be overriden by the organizer, if they fill in a session submission themselves.

**Comment by maze-runnar** [comment]
> This can be overriden by the organizer, if they fill in a session submission themselves.

Means organizer can set email to null?

**Comment by maze-runnar** [comment]
server is giving 422 now:

```
{"errors": [{"status": 422, "source": {"pointer": "/data/attributes"}, "title": "Unprocessable Entity", "detail": "Missing required fields ['email']"}], "jsonapi": {"version": "1.0"}}
```

**Comment by iamareebjamal** [comment]
Yes, only organizers can override email to null

**Pull request merged**